### PR TITLE
Node provider history + rewards fixes

### PR DESCRIPTION
**Pull request by mattdf**

### CODE REVIEW

File: src/ComputeRegistry.sol [hunk 1]
```diff
@@ -71,9 +71,17 @@ contract ComputeRegistry is IComputeRegistry, AccessControlEnumerable {
         returns (uint256)
     {
         address existingProvider = nodeProviderMap[subkey];
-        require(
-            existingProvider == address(0), "ComputeRegistry: node already exists or registered to a different provider"
-        );
+        if (existingProvider != address(0)) {
+            require(existingProvider == provider, "ComputeRegistry: node has already registered with another provider");
+            // check if this node already exists in this providers list
+            uint256 nodeIndex = nodeSubkeyToIndex.get(subkey);
```
Comment (manveerxyz) [inline src/ComputeRegistry.sol:77]: there exists an edge case where this could revert and prevent the re-addition of a node -- might be worth using `tryGet` instead?

```solidity
contract PrimeNetworkTest is Test {

    // ...

    function test_reAddRemovedNodeReverts() public {
        // register + whitelist provider
        addProvider(provider_good1);
        whitelistProvider(provider_good1);

        // add an inactive node (default isInactive = true -> removable)
        addNode(provider_good1, node_good1, node_good1_sk);

        // remove it — this clears nodeSubkeyToIndex but *not* nodeProviderMap
        removeNode(provider_good1, node_good1);

        // re-add the same subkey
        addNode(provider_good1, node_good1, node_good1_sk);

        // should have successfully re-added the node
        assertEq(computeRegistry.getNodeProvider(node_good1), provider_good1);
    }
}
```

File: test/PrimeNetwork.t.sol [hunk 2]
```diff
@@ -937,4 +938,63 @@ contract PrimeNetworkTest is Test {
...
+
+        validateNode(provider_good1, node_good1);
+
+        assertEq(computeRegistry.getNode(provider_good1, node_good1).subkey, node_good1);
+
+        nodeJoin(domain, pool, provider_good1, node_good1);
+
+        assertEq(isNodeInPool(pool, node_good1), true);
+
+        IRewardsDistributor rewardsDistributor = IRewardsDistributor(computePool.getRewardDistributorForPool(pool));
```
Comment (manveerxyz) [inline test/PrimeNetwork.t.sol:961]: wonder if it's worth testing the rewards claiming on all the different RewardsDistributor types since they all have different logic to calculate rewards? as a sanity check.

Comment (mattdf) [reply test/PrimeNetwork.t.sol:961]: it's hard to do here because the rewards factory is set when the compute pool contract is deployed (rather than when a pool is created). all the different claiming flows are tested in their individual test contracts, but to test them all here would require a lot of changes to how the factory setup works. we should def simplify/rework that too later but i think better as a separate task/PR.

File: src/ComputeRegistry.sol (hunk 1 above)
Comment (mattdf) [reply src/ComputeRegistry.sol]: good catch! mixed up the default key behaviour of EnumerableMap with the normal mapping - switched to tryGet and added the test.

File: test/PrimeNetwork.t.sol (hunk 2 above)
Comment (manveerxyz) [reply test/PrimeNetwork.t.sol:961]: makes sense!

**Review by manveerxyz — APPROVED** [review]
